src/globus_sdk/types.py

```diff
@@ -0,0 +1,12 @@
+from array import array
+from typing import Union
+
+from typing_extensions import Protocol
+
+# From: https://docs.python.org/3/glossary.html#term-bytes-like-object
+BytesLike = Union[array, bytes, bytearray, memoryview]
+
+
+class ToStr(Protocol):
+    def __str__(self) -> str:
+        ...
```

Comment:
Since there is a default definition of `__str__ = __repr__`, is this protocol significantly different from `Any`?

Does this enforce that the object must define an explicit, non-default `__str__`?